### PR TITLE
Prevent push spam clicking

### PR DESCRIPTION
**Pull request by developer239**
Inspired by [53fdbe6](https://github.com/wix/react-native-navigation/commit/53fdbe6315850a9167f5f2f06f68998395292a5f)

This was a major problem on slower devices. :)

### CODE REVIEW

**Comment by yershalom** [comment]
Can one of the admins verify this patch?

**Comment by filipef101** [comment]
works fine, but can you make it for other things than push?

**Comment by developer239** [comment]
@filipef101 I can grant you access to the repository if you want. :) I can't get to this before Sunday. :/

**Comment by filipef101** [comment]
I solved this by throtling my buttons, just commented as this doesnt fully solve the problem afaik
Also this won't be accepted as it is not a react native or react native navigation problem but multithreading ;)
At most I think it could be something optional. Also this library is mostly dead, and the issues... At least if one's doing a multiplatform app, so many pains in the arse

**Comment by developer239** [comment]
@filipef101 The sad thing is that this is still one of the most useful react-native navigations today. 😄

**Comment by filipef101** [comment]
yeah, sad indeed, but wix or wix devs are not to blame IMO
I think v2 will solve everything hopefully :)

File: src/Screen.js [hunk 1]
```diff
@@ -177,6 +187,17 @@ class Navigator {
     }
   }
 
+  _checkLastAction(params) {
+    if (Date.now() - this._lastAction.timestamp < 800
```
Comment (no23reason) [inline src/Screen.js]: How did you choose this value? What if the pushed screen takes longer to render?

Semi related: is there a real scenario where pushing the same screen multiple times was actually desired? I mean, would it break something if you omitted the timestamp altogether?

**Pull request closed without merging**